File: VB/WindowsApplication1/MyColorEdit/MyColorEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors

Namespace CustomEditors
	Public Class MyColorEdit
		Inherits ColorEdit
		Shared Sub New()
			RepositoryItemMyColorEdit.Register()
		End Sub
		Public Sub New()
		End Sub

		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemMyColorEdit.EditorName
			End Get
		End Property

		Protected Overrides Function CreatePopupForm() As DevExpress.XtraEditors.Popup.PopupBaseForm
			Return New MyPopupColorEditForm(Me)
		End Function
	End Class
End Namespace

File: CS/WindowsApplication1/MyColorEdit/MyColorEdit.cs
using System;
using DevExpress.XtraEditors;

namespace CustomEditors
{
    public class MyColorEdit : ColorEdit
    {
        static MyColorEdit()
        {
            RepositoryItemMyColorEdit.Register();
        }
        public MyColorEdit() { }

        public override string EditorTypeName
        {
            get { return RepositoryItemMyColorEdit.EditorName; }
        }

        protected override DevExpress.XtraEditors.Popup.PopupBaseForm CreatePopupForm()
        {
            return new MyPopupColorEditForm(this);
        }
    }
}
